proto/proto-backplane-grpc/src/main/proto/deephaven/proto/table.proto

```diff
@@ -418,30 +457,40 @@ message UpdateByRequest {
         }
 
         message UpdateByEma {
-          message UpdateByEmaOptions {
-            // If unset, defaults to server-provided defaults.
-            optional BadDataBehavior on_null_value = 1;
-
-            // If unset, defaults to server-provided defaults.
-            optional BadDataBehavior on_nan_value = 2;
+          // If unset, defaults to server-provided defaults.
+          UpdateByEmaOptions options = 1;
+          UpdateByEmaTimescale timescale = 2;
+        }
 
-            // If unset, defaults to server-provided defaults.
-            optional BadDataBehavior on_null_time = 3;
+        message UpdateByEms {
+          // If unset, defaults to server-provided defaults.
+          UpdateByEmaOptions options = 1;
+          UpdateByEmaTimescale timescale = 2;
+        }
 
-            // If unset, defaults to server-provided defaults.
-            optional BadDataBehavior on_negative_delta_time = 4;
+        message UpdateByEmMin {
+          // If unset, defaults to server-provided defaults.
+          UpdateByEmaOptions options = 1;
+          UpdateByEmaTimescale timescale = 2;
+        }
 
-            // If unset, defaults to server-provided defaults.
-            optional BadDataBehavior on_zero_delta_time = 5;
+        message UpdateByEmMax {
+          // If unset, defaults to server-provided defaults.
+          UpdateByEmaOptions options = 1;
+          UpdateByEmaTimescale timescale = 2;
+        }
 
-            // If unset, defaults to server-provided defaults.
-            MathContext big_value_context = 6;
-          }
+        message UpdateByEmStd {
           // If unset, defaults to server-provided defaults.
           UpdateByEmaOptions options = 1;
```

Comment:
Optional? --- basically same question in all places.

Reply:
Good question.  This `UpdateByEmaOptions` structure is required but child members of it are optional and will be filled with server defaults when not supplied.

Reply:
I rethought this and these (and the others) are truly optional and should be marked `optional` in the protobuf files.